lib/smart_proxy_dynflow_core/launcher.rb

```diff
@@ -47,7 +49,7 @@ def self.route_mapping(rack_builder)
 
     def install_usr1_trap
       trap(:USR1) do
-        Log.instance.roll_log
+        Log.reopen
```

Comment:
Perhaps, instead of complaining about quality of old code, you might want to spend the time testing the stuff a bit more so that this one is top class :) This currently gives:

```
# pkill -f -USR1 bin/smart_proxy_dynflow_core
 2019-04-24T16:53:11  [E] ThreadError: can't be called from trap context
	/home/inecas/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/logging-2.2.2/lib/logging/utils.rb:159:in `synchronize'
```

Seems like there is a reason for the code in smart-proxy not being that straight forward https://github.com/theforeman/smart-proxy/blob/8652a7e5c63009cae026f6350178583dcadc5f34/lib/proxy/log_buffer/decorator.rb#L25

Reply:
Nice, thanks for the insight. Amended the very same approach.

Reply:
Do we even need the usr1 trap? Originally this was here because the log rotation was done by an external process. If I understand this correctly log rotation will be handled by the process itself so it doesn't need to be notified from the outside. Or am I missing something?

Reply:
No it was not decided yet and I am actually leaning towards keeping logrotate after some more thoughts. This is here only because logging gem supports it so there is an option e.g. for gem installations. However I want to make rolling optional now.